### PR TITLE
Add fuzzy query for log content.

### PR DESCRIPTION
**Pull request by wu-sheng**
Log content fuzzy query is a risky operation. What do you think about this?

### CODE REVIEW

**Comment by EvanLjp** [comment]
Log queries are much more complex than other existing queries in SkyWalking now, such as more data size and need custom Tokenizer in es.

**Comment by wu-sheng** [comment]
That is another story about the stop word, word analyzer, etc. But none of them relate to the query abstract implementation and protocol.
Let's discuss one thing at a time.

**Comment by kezhenxu94** [comment]
I'm big +1 to have this functionality, the performance concern will be discussed in specific implementations though.

**Review by kezhenxu94 — APPROVED** [review]

**Comment by dmsolr** [comment]
I'm +1 at first of all.
I am thinking about the implementation of storage plugins. It seems that not very friendly except ES.

**Review by dmsolr — APPROVED** [review]

**Comment by wu-sheng** [comment]
> I am thinking about the implementation of storage plugins. It seems that not very friendly except ES.

@dmsolr Yes, I am thinking about this too. How about we add a response text when the implementation doesn't support that, they could add description for the UI.

**Review by EvanLjp — APPROVED** [review]

**Review by arugal — APPROVED** [review]
I think this is a very useful functionality.

File: log.graphqls [hunk 1]
```diff
@@ -82,5 +84,6 @@ enum ContentType {
 }
 
 extend type Query {
+    supportQueryLogsByKeywords: Boolean!
```
Comment (wu-sheng) [inline log.graphqls:87]: @arugal @kezhenxu94 @EvanLjp @dmsolr What about this? And I put keywords as List.

**Comment by EvanLjp** [comment]
Maybe need to be enhanced the feature. We need not only `grep` but `grep -v` in the original log query. `excludeKeywordsOfContent ` may need to be added.

File: log.graphqls [hunk 2]
```diff
@@ -55,6 +55,10 @@ input LogQueryCondition {
     state: LogState!
     paging: Pagination!
     tags: [LogTag!]
+    # Fuzzy query conditions for the log content.
+    # Use these 2 keyword related condition, when supportQueryLogsByKeywords = TRUE.
+    keywordsOfContent: [!String]
+    excludingKeywordsOfContent: [!String]
```
Comment (wu-sheng) [inline log.graphqls:61]: @EvanLjp You ask for this, right?

Comment (EvanLjp) [reply log.graphqls:61]: yes

**Pull request merged**